packages/serverpod_shared/lib/src/config.dart

```diff
@@ -114,6 +114,14 @@ class ServerConfig {
 
     return str;
   }
+
+  /// Get [Uri] from publicHost, port and publicScheme.
+  Uri toUri() {
+    if (publicScheme == 'http') {
+      return Uri.http('$publicHost:$publicPort');
+    }
+    return Uri.https(publicHost);
```

Comment:
Here we also need to specify the port to keep things consistent.

We can also ignore the port for http 80, and https 443, as those are defaults.